### PR TITLE
[FIX] [Livechat] Unread messages badge

### LINKED ISSUE #25529: A message sent by an agent in a livechat room is automatically marked as read
### Description:

A message sent by an agent in a livechat room is automatically marked as read

### Steps to reproduce:

1. Open room in Rocket Chat client
2. Send message to livechat from agent

### Expected behavior:

The message has been sent, while it has the status of unread until the livechat guest reads it

### Actual behavior:

Message automatically marked as read

### Server Setup Information:

- Version of Rocket.Chat Server: 4.6.3
- NodeJS Version: 14.18.3
- MongoDB Version: 4.2.19

### PR DESCRIPTION
**Pull request by tiagoevanp**
## Proposed changes (including videos or screenshots)

OC-169

Even if the page was reopening, Livechat will inform unread messages

## Issue(s)

close #25529

### CODE REVIEW

**Review by aleksandernsilva — DISMISSED** [review]
Could you add e2e tests to prevent this issue from happening again?

File: apps/meteor/tests/e2e/omnichannel-livechat.spec.ts [hunk 1]
```diff
@@ -16,33 +16,27 @@ const newUser = {
...
-
-				poAuxContext = await createAuxContext(browser, 'user1-session.json');
-			});
-
-			test.afterAll(async () => {
-				await poAuxContext.page.close();
-			});
-
-			test('expect message is received from agent and user ', async ({ page }) => {
+			test('expect message is sended by livechat', async () => {
```
Comment (MartinSchoeler) [inline apps/meteor/tests/e2e/omnichannel-livechat.spec.ts:39]:
```suggestion
			test('Expect message to be sent by livechat', async () => {
```

😃

File: apps/meteor/tests/e2e/omnichannel-livechat.spec.ts [hunk 2]
```diff
@@ -52,20 +46,33 @@ test.describe('Livechat', () => {
 				await expect(page.locator('div >>text="this_a_test_message_from_user"')).toBeVisible();
 			});
 
-			test('expect after user close live chat screen dont show messages', async ({ page }) => {
+			test('expect message is received by agent', async () => {
```
Comment (MartinSchoeler) [inline apps/meteor/tests/e2e/omnichannel-livechat.spec.ts:49]:
```suggestion
			test('expect message to be received by agent', async () => {
```

File: apps/meteor/tests/e2e/omnichannel-livechat.spec.ts [hunk 3]
```diff
@@ -52,20 +46,33 @@ test.describe('Livechat', () => {
...
+				await poAuxContext.poHomeChannel.sidenav.openChat(newUser.name);
+				await expect(poAuxContext.poHomeChannel.content.lastUserMessage).toBeVisible();
+				await expect(poAuxContext.poHomeChannel.content.lastUserMessage).toContainText('this_a_test_message_from_user');
+			});
+		});
+
+		test.describe('Send message to livechat costumer', () => {
+			test('expect message is sended by agent', async () => {
```
Comment (MartinSchoeler) [inline apps/meteor/tests/e2e/omnichannel-livechat.spec.ts:57]:
```suggestion
			test('Expect message to be sent by agent', async () => {
```

File: apps/meteor/tests/e2e/omnichannel-livechat.spec.ts [hunk 4]
```diff
@@ -52,20 +46,33 @@ test.describe('Livechat', () => {
...
+				await poAuxContext.poHomeChannel.content.sendMessage('this_a_test_message_from_agent');
+				await expect(page.locator('div >>text="this_a_test_message_from_agent"')).toBeVisible();
+			});
+
+			test("expect after user minimize livechat screen don't show composer", async () => {
```
Comment (MartinSchoeler) [inline apps/meteor/tests/e2e/omnichannel-livechat.spec.ts:62]:
```suggestion
			test("Expect when user minimizes the livechat screen, the composer should be hidden", async () => {
```

File: apps/meteor/tests/e2e/omnichannel-livechat.spec.ts [hunk 5]
```diff
@@ -52,20 +46,33 @@ test.describe('Livechat', () => {
...
-	test.describe('Verify message is received', () => {
-		test.use({ storageState: 'user1-session.json' });
-		test('expect message is received from user', async ({ page }) => {
-			await page.goto('/');
-			const poHomeChannel = new HomeChannel(page);
-			await poHomeChannel.sidenav.openChat(newUser.name);
-			await expect(poHomeChannel.content.lastUserMessage).toBeVisible();
-			await expect(poHomeChannel.content.lastUserMessage).toContainText('this_a_test_message_from_user');
+
+			test('expect message is received by minimized livechat', async () => {
```
Comment (MartinSchoeler) [inline apps/meteor/tests/e2e/omnichannel-livechat.spec.ts:67]:
```suggestion
			test('expect message to be received by minimized livechat', async () => {
```

File: apps/meteor/tests/e2e/omnichannel-livechat.spec.ts [hunk 6]
```diff
@@ -52,20 +46,33 @@ test.describe('Livechat', () => {
...
+				await poAuxContext.poHomeChannel.content.sendMessage('this_a_test_message_again_from_agent');
+				await expect(poLiveChat.unreadMessagesBadge).toBeVisible();
+			});
+
+			test('expect unread messages be visible after a reload', async () => {
```
Comment (MartinSchoeler) [inline apps/meteor/tests/e2e/omnichannel-livechat.spec.ts:72]:
```suggestion
			test('expect unread messages to be visible after a reload', async () => {
```

**Review by MartinSchoeler — CHANGES_REQUESTED** [review]
Just some grammar suggestions on the tests nomenclature

**Review by MartinSchoeler — DISMISSED** [review]

File: apps/meteor/client/components/Omnichannel/modals/CloseChatModal.tsx [hunk 7]
```diff
@@ -103,7 +103,7 @@ const CloseChatModal = ({
 			<Modal.Footer>
 				<Modal.FooterControllers>
 					<Button onClick={onCancel}>{t('Cancel')}</Button>
-					<Button type='submit' disabled={cannotSubmit} primary>
+					<Button role='button' type='submit' disabled={cannotSubmit} primary>
```
Comment (tiagoevanp) [inline apps/meteor/client/components/Omnichannel/modals/CloseChatModal.tsx:106]:
```suggestion
					<Button type='submit' disabled={cannotSubmit} primary>
```

**Review by aleksandernsilva — APPROVED** [review]

File: packages/livechat/src/i18n/en.json [hunk 8]
```diff
@@ -82,6 +82,8 @@
         "the_controller_of_your_personal_data_is_company_na": "The controller of your personal data is [Company Name], with registered office at [Company Address]. To start the chat you agree that your personal data shall be processed and trasmitted in accordance with the General Data Protection Regulation (GDPR).",
 				"transcript_success": "Transcript sent",
         "type_your_message_here": "Type your message here",
+				"unread_messages_count_one": "{{count}} unread message",
+				"unread_messages_count_other": "{{count}} unread messages",
```
Comment (ggazzo) [inline packages/livechat/src/i18n/en.json:86]: 👀

**Review by ggazzo — APPROVED** [review]

**Pull request merged**